dss-apps/dss-datamodel-center/dss-datamodel-dimension-management/src/main/java/com/webank/wedatasphere/dss/datamodel/dimension/service/impl/DimensionServiceImpl.java

```diff
@@ -0,0 +1,214 @@
+            throw new DSSDatamodelCenterException(ErrorCode.DIMENSION_UPDATE_ERROR.getCode(), "dimension name can not  referenced");
+        }
+
+
+        //当更新名称时
+        if (!StringUtils.equals(vo.getName(), org.getName())) {
+            int repeat = getBaseMapper().selectCount(Wrappers.<DssDatamodelDimension>lambdaQuery().eq(DssDatamodelDimension::getName, vo.getName()));
+            if (repeat > 0) {
+                LOGGER.error("errorCode : {}, dimension name can not repeat ", ErrorCode.DIMENSION_UPDATE_ERROR.getCode());
+                throw new DSSDatamodelCenterException(ErrorCode.DIMENSION_UPDATE_ERROR.getCode(), "dimension name can not repeat ");
+            }
+        }
+
+        String orgFieldIdentifier = org.getFieldIdentifier();
+        //当更新标识时
+        if (!StringUtils.equals(vo.getFieldIdentifier(), orgFieldIdentifier)) {
+            int repeat = getBaseMapper().selectCount(Wrappers.<DssDatamodelDimension>lambdaQuery().eq(DssDatamodelDimension::getFieldIdentifier, vo.getFieldIdentifier()));
+            if (repeat > 0) {
+                LOGGER.error("errorCode : {}, dimension field identifier can not repeat", ErrorCode.DIMENSION_UPDATE_ERROR.getCode());
+                throw new DSSDatamodelCenterException(ErrorCode.DIMENSION_UPDATE_ERROR.getCode(), "dimension field identifier can not repeat ");
+            }
+        }
+        DssDatamodelDimension updateOne = modelMapper.map(vo,DssDatamodelDimension.class);
+        updateOne.setUpdateTime(new Date());
+        getBaseMapper().update(updateOne, Wrappers.<DssDatamodelDimension>lambdaUpdate().eq(DssDatamodelDimension::getId,id));
+
+        //同步atlas
+        UpdateModelTypeResult updateModelTypeResult = assertsSyncService.syncUpdateModel(
+                new UpdateModelEvent(this
+                        ,DataModelSecurityContextHolder.getContext().getDataModelAuthentication().getUser()
+                        ,vo.getFieldIdentifier()
+                        ,orgFieldIdentifier
+                        ,ClassificationConstant.DIMENSION));
+        return 1;
+    }
+
+
+    @Override
+    @Transactional(rollbackFor = Exception.class)
+    public int deleteDimension(Long id) throws ErrorException{
+
+        DssDatamodelDimension dssDatamodelDimension = getBaseMapper().selectById(id);
+        if (dssDatamodelDimension == null){
+            throw new DSSDatamodelCenterException(ErrorCode.DIMENSION_DELETE_ERROR.getCode(), "dimension id " +id +" not exists");
+        }
+        //校验引用情况
+        if(datamodelReferencService.dimensionReferenceCount(dssDatamodelDimension.getName())>0||datamodelReferencService.dimensionReferenceCount(dssDatamodelDimension.getFieldIdentifier())>0){
+              throw new DSSDatamodelCenterException(ErrorCode.DIMENSION_DELETE_ERROR.getCode(), "dimension id " +id +" has referenced");
+        }
+        getBaseMapper().deleteById(id);
+        //同步资产
+        assertsSyncService.syncDeleteModel(new DeleteModelEvent(this
```

Comment:
publisher.publishEvent(new DeleteModelEvent(this
                ,DataModelSecurityContextHolder.getContext().getDataModelAuthentication().getUser()
                ,dssDatamodelDimension.getFieldIdentifier()
                ,ClassificationConstant.DIMENSION));